components/dashboard/TextArea/index.tsx

```diff
@@ -49,14 +50,39 @@ const TextArea = ({ shiftRight, setShiftRight }: TextAreaProps) => {
       ).withConverter(notesConverter)
   );
 
-  const saveNoteChanges = async (noteId: string) => {
-    const lsInput = localStorage.getItem("editorContent");
-    const lsTitle = localStorage.getItem("editorTitle");
-    if (!isSynced && selectedNoteId && lsInput && lsTitle) {
+  const addToStore = async (
+    key: string,
+    value: { editorTitle: string; editorContent: string }
+  ) => {
+    set(key, value);
+  };
+
+  const getFromStore = async (
+    selectedNoteId: string
+  ): Promise<{ editorTitle: string; editorContent: string } | null> => {
+    if (!selectedNoteId) return null;
+    return (
+      (await get(selectedNoteId)) || {
+        editorTitle: "Untitled",
+        editorContent: "",
+      }
+    );
+  };
+
+  const saveNoteChanges = async (noteId: string): Promise<void> => {
+    if (!selectedNoteId) return;
+
+    const idbObject = await getFromStore(selectedNoteId);
+
+    if (!idbObject) return;
+
+    const { editorContent, editorTitle } = idbObject;
```

Comment:
What if editorTitle is ""?

Reply:
Is that supposed to cause problems?

Reply:
Hmm, maybe?
I noticed that editorTitle is used in updateNote function. We should, at the lowest level of interaction with firestore, add a check for setting the value to Untitled if null.

Reply:
That makes sense, thanks